chia/wallet/nft_wallet/nft_off_chain.py

```diff
@@ -0,0 +1,108 @@
+import logging
+import sys
+from pathlib import Path
+from typing import Optional
+
+import aiohttp
+
+from chia.types.blockchain_format.program import Program
+from chia.types.blockchain_format.sized_bytes import bytes32
+from chia.util.default_root import DEFAULT_ROOT_PATH
+from chia.util.hash import std_hash
+from chia.wallet.nft_wallet.nft_info import NFTCoinInfo
+from chia.wallet.nft_wallet.uncurry_nft import UncurriedNFT
+
+PREFIX_HASH_LENGTH = 3
+CACHE_PATH_KEY = "nft_metadata_cache_path"
+DEFAULT_PATH = DEFAULT_ROOT_PATH / "nft_cache"
+
+log = logging.getLogger(__name__)
+
+
+async def fetch_off_chain_metadata(nft_coin_info: NFTCoinInfo, ignore_size_limit: bool) -> Optional[str]:
+    uncurried_nft: Optional[UncurriedNFT] = UncurriedNFT.uncurry(*nft_coin_info.full_puzzle.uncurry())
+    if uncurried_nft is None:
+        log.error(f"Cannot fetch off-chain metadata, {nft_coin_info.nft_id} is not a NFT.")
+        return None
+    for uri in uncurried_nft.meta_uris.as_python():  # pylint: disable=E1133
+        async with aiohttp.ClientSession() as session:
+            async with session.get(str(uri, "utf-8")) as response:
+                if response.status == 200:
+                    text = await response.text()
```

Comment:
wouldn't this load all fetched file content into memory?

Reply:
let's say metadata file is 100GB in size, wouldn't this try to load the whole file content into `text` variable?

Reply:
What is the solution for this in Python? I was thinking use content-length to check the file size but it is not reliable.

Reply:
you can use the streaming API: https://docs.aiohttp.org/en/stable/streams.html

for example:
```python
async with aiohttp.ClientSession() as session:
    async with session.get(str(uri, "utf-8")) as response:
        if response.status == 200:
            async for chunk in response.content.iter_chunked(1024):
                file_obj.write(chunk)
```